src/components/ImageGallery.vue

```diff
@@ -0,0 +1,501 @@
+          const id = plot.identifier;
+          const thumbnail = this.getThumbnailForPlot(plot, this.entry.thumbnails);
+          let thumbnailURL = undefined;
+          let mimetype = '';
+          if (thumbnail) {
+            thumbnailURL = this.getImageURLFromS3(this.envVars.API_LOCATION, {
+              id,
+              datasetId: this.datasetId,
+              datasetVersion: this.datasetVersion,
+              file_path: thumbnail.dataset.path,
+            });
+            mimetype = thumbnail.mimetype.name;
+          }
+          const plotAnnotation = plot.datacite;
+          const filePathPrefix = `${this.envVars.API_LOCATION}/s3-resource/${this.datasetId}/${this.datasetVersion}/files/`;
+          const sourceUrl = filePathPrefix + plot.dataset.path;
+
+          const metadata = JSON.parse(
+            plotAnnotation.supplemental_json_metadata.description
+          );
+
+          let supplementalData = [];
+          if (plotAnnotation.isDescribedBy) {
+            supplementalData.push({
+              url: filePathPrefix + plotAnnotation.isDescribedBy.path
+            });
+          }
+
+          const resource = {
+            dataSource: {url: sourceUrl},
+            metadata,
+            supplementalData
+          }
+
+          let action = {
+            label: capitalise(this.label),
+            resource: resource,
+            title: "View plot",
+            type: "Plot",
+            discoverId: this.discoverId,
+            version: this.datasetVersion,
+          };
+          this.items['Plots'].push({
+            id,
+            title: baseName(filePath),
+            type: "Plot",
+            thumbnail: thumbnailURL,
+            userData: action,
+            hideType: true,
+            mimetype
+          });
+        });
+      }
+    },
+    createScaffoldItems: function () {
+      if (this.entry.scaffolds) {
+        let index = 0;
+        this.entry.scaffolds.forEach((scaffold) => {
+          const filePath = scaffold.dataset.path;
+          const id = scaffold.identifier;
+          const thumbnail = this.getThumbnailForScaffold(
+            scaffold,
+            this.entry.scaffoldViews,
+            this.entry.thumbnails,
+            index
+          );
+          let mimetype = '';
+          let thumbnailURL = undefined;
+          if (thumbnail) {
+            thumbnailURL = this.getImageURLFromS3(this.envVars.API_LOCATION, {
+              id,
+              datasetId: this.datasetId,
+              datasetVersion: this.datasetVersion,
+              file_path: thumbnail.dataset.path,
+            });
+            mimetype = thumbnail.mimetype.name;
+          }
+          let action = {
+            label: capitalise(this.label),
+            resource: `${this.envVars.API_LOCATION}s3-resource/${this.datasetId}/${this.datasetVersion}/files/${filePath}`,
+            title: "View 3D scaffold",
+            type: "Scaffold",
+            discoverId: this.datasetId,
```

Comment:
Missing context cards here.

(see https://github.com/ABI-Software/map-sidebar/blob/061b738c42866214542c117b17512ae081e557b7/src/components/DatasetCard.vue#L151)

Reply:
I will add that in.